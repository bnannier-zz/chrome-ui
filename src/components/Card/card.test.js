import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { Card } from "../";

describe("Card Tests", () => {
  it("Default", function() {
    const wrapper = shallow(<Card>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Outlined", function() {
    const wrapper = shallow(<Card outlined>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Primary Action", function() {
    const wrapper = shallow(<Card primaryAction>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Media", function() {
    const wrapper = shallow(<Card media>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Media Square", function() {
    const wrapper = shallow(<Card mediaSquare>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Media 169", function() {
    const wrapper = shallow(<Card media169>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Media Content", function() {
    const wrapper = shallow(<Card mediaContent>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Actions", function() {
    const wrapper = shallow(<Card actions>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Actions Full Bleed", function() {
    const wrapper = shallow(<Card actionsFullBleed>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Action Buttons", function() {
    const wrapper = shallow(<Card actionButtons>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Action Icons", function() {
    const wrapper = shallow(<Card actionIcons>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Action", function() {
    const wrapper = shallow(<Card action>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Action Button", function() {
    const wrapper = shallow(<Card actionButton>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Action Icon", function() {
    const wrapper = shallow(<Card actionIcon>test</Card>);
    expect(wrapper).toMatchSnapshot();
  });
});
