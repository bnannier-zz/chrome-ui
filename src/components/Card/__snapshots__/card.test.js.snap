// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests Action 1`] = `
<Base
  action={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": true,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Action Button 1`] = `
<Base
  actionButton={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": true,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Action Buttons 1`] = `
<Base
  actionButtons={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": true,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Action Icon 1`] = `
<Base
  actionIcon={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": true,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Action Icons 1`] = `
<Base
  actionIcons={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": true,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Actions 1`] = `
<Base
  actions={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": true,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Actions Full Bleed 1`] = `
<Base
  actionsFullBleed={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": true,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Default 1`] = `
<Base>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Media 1`] = `
<Base
  media={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": true,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Media 169 1`] = `
<Base
  media169={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": true,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Media Content 1`] = `
<Base
  mediaContent={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": true,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Media Square 1`] = `
<Base
  mediaSquare={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": true,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Outlined 1`] = `
<Base
  outlined={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": true,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": undefined,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;

exports[`Card Tests Primary Action 1`] = `
<Base
  primaryAction={true}
>
  <div
    className={
      Array [
        "mdc-card",
        Object {
          "mdc-card--outlined": undefined,
          "mdc-card__action": undefined,
          "mdc-card__action--button": undefined,
          "mdc-card__action--icon": undefined,
          "mdc-card__action-buttons": undefined,
          "mdc-card__action-icons": undefined,
          "mdc-card__actions": undefined,
          "mdc-card__actions--full-bleed": undefined,
          "mdc-card__media": undefined,
          "mdc-card__media--16-9": undefined,
          "mdc-card__media-content": undefined,
          "mdc-card__primary-action": true,
          "mdc-mdc-card__media--square": undefined,
        },
      ]
    }
  >
    test
  </div>
</Base>
`;
